sdk/loadtestservice/load-testing-rest/src/getFileValidationPoller.ts

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AbortController, AbortSignalLike } from "@azure/abort-controller";
+import { CancelOnProgress, OperationState, SimplePollerLike } from "@azure/core-lro";
+import { FileUploadAndValidatePoller, PolledOperationOptions } from "./models";
+import { AzureLoadTestingClient } from "./clientDefinitions";
+import { TestGetFile200Response, TestUploadFile201Response } from "./responses";
+import { isUnexpected } from "./isUnexpected";
+import { sleep } from "./util/LROUtil";
+
+/**
+ * Uploads a file and creates a poller to poll for validation.
+ * @param client - The Load Testing client.
+ * @param options - The operation options.
+ * @returns A poller which can be called to poll until completion of the job.
+ */
+export async function getFileValidationPoller(
+  client: AzureLoadTestingClient,
+  fileUploadResult: TestUploadFile201Response,
+  polledOperationOptions: PolledOperationOptions = {}
+): Promise<FileUploadAndValidatePoller> {
+  // get filename and testid from initial response
+  const fileName = fileUploadResult.body.fileName;
+  const requestUrl = fileUploadResult.request.url;
+  const testId = requestUrl.substring(
+    requestUrl.indexOf("tests/") + 6,
+    requestUrl.lastIndexOf("/files")
+  );
+  type Handler = (state: OperationState<TestGetFile200Response>) => void;
+
+  const state: OperationState<TestGetFile200Response> = {
+    status: "notStarted",
+  };
+
+  const progressCallbacks = new Map<symbol, Handler>();
+  const processProgressCallbacks = async (): Promise<void> =>
+    progressCallbacks.forEach((h) => h(state));
+  let resultPromise: Promise<TestGetFile200Response> | undefined;
+  let cancelJob: (() => void) | undefined;
+  const abortController = new AbortController();
+  const currentPollIntervalInMs = polledOperationOptions.updateIntervalInMs ?? 2000;
+
+  const poller: SimplePollerLike<OperationState<TestGetFile200Response>, TestGetFile200Response> = {
+    async poll(_options?: { abortSignal?: AbortSignalLike }): Promise<void> {
+      if (_options?.abortSignal?.aborted) {
+        state.status = "canceled";
+        state.error = new Error("The operation was aborted.");
```

Comment:
The operation itself wasn't aborted:
```suggestion
        state.error = new Error("The polling was aborted.");
```

Reply:
Updated.